v5.12.7 之后的版本都无法在 pnpm hoist:false 下通过 vite 编译

### Reproduction link

[https://github.com/liuweiGL/antd-babel_runtime](https://github.com/liuweiGL/antd-babel_runtime)

### Steps to reproduce

1. git clone https://github.com/liuweiGL/antd-babel_runtime
2. cd antd-babel_runtime
3. pnpm i
4. pnpm dev

### What is expected?

能正常运行

### What is actually happening?

编译报错：
```
VITE v5.1.4  ready in 148 ms

  ➜  Local:   http://localhost:9090/
  ➜  Network: use --host to expose
  ➜  press h   enter to show help
✘ [ERROR] Could not resolve "@babel/runtime/helpers/esm/classCallCheck"

    ../../node_modules/.pnpm/antd@5.14.2_react-dom@18.2.0_react@18.2.0/node_modules/antd/es/alert/ErrorBoundary.js:3:28:      3 │ import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
        ╵                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@babel/runtime/helpers/esm/classCallCheck" as external to exclude it from
  the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "@babel/runtime/helpers/esm/toConsumableArray"

    ../../node_modules/.pnpm/antd@5.14.2_react-dom@18.2.0_react@18.2.0/node_modules/antd/es/watermark/index.js:3:31:
      3 │ import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@babel/runtime/helpers/esm/toConsumableArray" as external to exclude it
  from the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "@babel/runtime/helpers/esm/toConsumableArray"

    ../../node_modules/.pnpm/antd@5.14.2_react-dom@18.2.0_react@18.2.0/node_modules/antd/es/cascader/index.js:3:31:
      3 │ import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@babel/runtime/helpers/esm/toConsumableArray" as external to exclude it
  from the bundle, which will remove this error and leave the unresolved path in the bundle.

✘ [ERROR] Could not resolve "@babel/runtime/helpers/esm/toConsumableArray"

    ../../node_modules/.pnpm/antd@5.14.2_react-dom@18.2.0_react@18.2.0/node_modules/antd/es/modal/useModal/index.js:3:31:
      3 │ import _toConsumableArray from "@babel/runtime/helpers/esm/toConsumableArray";
        ╵                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  You can mark the path "@babel/runtime/helpers/esm/toConsumableArray" as external to exclude it
  from the bundle, which will remove this error and leave the unresolved path in the bundle.
```

| Environment | Info |
| --- | --- |
| antd | 5.14.2 |
| React | 18.2.0 |
| System | Windows11 |
| Browser | Chrome 99.0.4844.51 |

---
https://github.com/ant-design/ant-design/commit/15c20511b48c743ad6d0b0627a391e2af4868181 提交不知道为什么把 @bable/runtime 移除了

<!-- generated by ant-design-issue-helper. DO NOT REMOVE -->

--- a/package.json
+++ b/package.json
@@ -119,6 +119,7 @@
     "@ant-design/cssinjs": "^1.18.4",
     "@ant-design/icons": "^5.3.0",
     "@ant-design/react-slick": "~1.0.2",
+    "@babel/runtime": "^7.23.9",
     "@ctrl/tinycolor": "^3.6.1",
     "@rc-component/color-picker": "~1.5.2",
     "@rc-component/mutate-observer": "^1.1.0",

